system-helper/flatpak-system-helper.c

```diff
@@ -1299,6 +1300,13 @@ handle_remove_local_ref (FlatpakSystemHelper   *object,
       return G_DBUS_METHOD_INVOCATION_HANDLED;
     }
 
+  ref = flatpak_decomposed_new_from_ref (arg_ref, &error);
```

Comment:
Unfortunately upon looking at this patch again a month later I think it's wrong, because it makes the `RemoveLocalRef` method error out when `arg_ref` does not begin with `app/` or `runtime/`, but there's a note on the `ref` arg of `flatpak_dir_remove_ref()` which says `/* NOTE: Not necessarily a app/runtime ref */`, and that's because we sometimes remove other refs, e.g. `appstream2/x86_64` or `ostree-metadata`.

Reply:
The `flatpak_is_app_runtime_or_appstream_ref()` helper would cover the cases where the ref has the prefix `app/`, `runtime/`, `appstream/`, or `appstream2/`. The only other possible refs I can think of are the `deploy/` ones and `ostree-metadata`.

We could, instead of trying to make an exhaustive list of Flatpak ref patterns, use libostree's `ostree_validate_rev`, which would give us a little more flexibility to add ref patterns in the future without needing to update the list here.

In any case, I guess this patch needs to miss the boat for 1.14.0, since I'm trying to get that release out in time for Fedora 37 which has beta freeze tomorrow.